.ci/scripts/backport.py

```
@@ -0,0 +1,523 @@
+    """
+    path = os.environ.get("GITHUB_EVENT_PATH")
+    if not path or not os.path.exists(path):
+        LOG.warning("GITHUB_EVENT_PATH not set or file missing; nothing to do")
+        return {}
+    with open(path, "r", encoding="utf-8") as f:
+        return json.load(f)
+
+
+def extract_pr(event: dict) -> PRInfo | None:
+    """Extract PR information from the event payload (single PR mode)."""
+    pr = event.get("pull_request")
+    return PRInfo(pr) if pr else None
+
+
+def list_prs(q_filter: str, since: dt.datetime) -> Iterable[dict[str, Any]]:
+    """Query the GH API with a filter to iterate over PRs updated after a given timestamp."""
+    q_date = since.strftime("%Y-%m-%d")
+    q = f"{q_filter} updated:>={q_date}"
+    LOG.debug(f"Fetch PRs with filter '{q}")
+    page = 1
+    while True:
+        result = gh_request(
+            path="/search/issues",
+            params={"q": f"{q}", "per_page": "100", "page": str(page)},
+        )
+        items = result.get("items", [])
+        if not items:
+            break
+        for it in items:
+            yield it
+        if len(items) < 100:
+            break
+        page += 1
+
+
+def repo_needs_pending_label(repo_labels: list[str]) -> bool:
+    return repo_labels.count(PENDING_LABEL) == 0
+
+
+def ensure_backport_pending_label() -> None:
+    """If the exact PENDING_LABEL string does not appear at least once, we create it."""
+    repo = CONFIG.repo
+    try:
+        existing = gh_request(path=f"/repos/{repo}/labels", params={"per_page": "100"})
+    except Exception as e:
+        LOG.info(f"Could not list labels; attempting direct create: {e}")
+        existing = []
+    LOG.debug(f"Repo has labels: {existing};")
+    names = [lbl.get("name", "") for lbl in existing]
+    if not repo_needs_pending_label(names):
+        return
+    try:
+        gh_request(method="POST", path=f"/repos/{repo}/labels", body={"name": PENDING_LABEL, "color": PENDING_LABEL_COLOR})
```

I think this should be wrapped by a method/function that says what this request is doing.
Example:

```
def add_repository_label(repository: str, name: str, color: str):
   gh_request(method="POST",  path=f"/repos/{repository}/labels", body={"name": name, "color": color})
```

Probably checking if the repo has already a label is not necessary. Just capture that exception if produces one error and return with no error. In case of any other error please raise or re-raise an error. It is an anti pattern on my opinion silencing errors and leave the code proceed unnotified in low level functions. It should the high level workflow function in charge of it.